### PR TITLE
chore: updated Husky to v6 and other template deps

### PR DESCRIPTION
**Pull request by pozil**
### What does this PR do?
This PR updates node dependencies in the new project template. In particular:
- it upgrades Husky from v4 to v6. This major upgrade requires new config files.
- it adds the new LWC/Aura/Lightning ESLint plugins and config (see #329)
- fixes [linting issue](https://github.com/forcedotcom/salesforcedx-vscode/issues/3312)

I used the [E-Bikes sample application](https://github.com/trailheadapps/ebikes-lwc) configuration as a reference.

### What issues does this PR fix or reference?
#329 and [linting issue](https://github.com/forcedotcom/salesforcedx-vscode/issues/3312)

### CODE REVIEW

File: packages/templates/src/generators/projectGenerator.ts [hunk 1]
```diff
@@ -111,6 +113,19 @@ export default class ProjectGenerator extends SfdxGenerator<ProjectOptions> {
 
     if (template === 'standard') {
       makeEmptyFolders(folderlayout, standardfolderarray);
+
+      // Add Husky config
+      makeEmptyFolders(folderlayout, [HUSKY_FOLDER]);
+      for (const file of huskyhookarray) {
+        this.fs.copyTpl(
+          this.templatePath(file),
```
Comment (brpowell) [inline packages/templates/src/generators/projectGenerator.ts:121]: Does this `src` path need to include the husky folder? Im getting this trying to create a project:

Screen Shot 2021-06-11 at 3 44 49 PM

Comment (pozil) [reply packages/templates/src/generators/projectGenerator.ts]: Thanks for catching this @brpowell. I wasn't testing this correctly.
`.husky` needs to be at the root of the project. I fixed this.

File: packages/templates/src/templates/project/package.json [hunk 2]
```diff
@@ -8,36 +8,40 @@
...
     "lint:lwc": "eslint **/lwc/**",
     "test": "npm run test:unit",
-    "test:unit": "sfdx-lwc-jest",
-    "test:unit:watch": "sfdx-lwc-jest --watch",
-    "test:unit:debug": "sfdx-lwc-jest --debug",
-    "test:unit:coverage": "sfdx-lwc-jest --coverage",
+    "test:unit": "sfdx-lwc-jest --skipApiVersionCheck",
+    "test:unit:watch": "sfdx-lwc-jest --skipApiVersionCheck --watch",
+    "test:unit:debug": "sfdx-lwc-jest --skipApiVersionCheck --debug",
+    "test:unit:coverage": "sfdx-lwc-jest --skipApiVersionCheck --coverage",
```
Comment (brpowell) [inline packages/templates/src/templates/project/package.json:14]: Do we want to keep these in even though we're bumping the major version this week?

Comment (pozil) [reply packages/templates/src/templates/project/package.json:14]: The platform version is hardcoded in `sfdx-lwc-jest` so bumping the API version in `sfdx-project.json` is not enough. We would need to also update the `sfdx-lwc-jest` version for each release unless we use that flag.

@pmdartus my understanding is that the LWC team will no longer release `sfdx-lwc-jest` new versions for each Salesforce release (npm indicates that the latest version is v0.10.4 for quite some time).
Are you ok to have the `--skipApiVersionCheck` flag in the new project template?

File: packages/templates/src/templates/project/package.json [hunk 3]
```diff
@@ -8,36 +8,40 @@
...
   "lint-staged": {
     "**/*.{cls,cmp,component,css,html,js,json,md,page,trigger,xml,yaml,yml}": [
       "prettier --write"
     ],
-    "**/{aura|lwc}/**": [
+    "**/{aura,lwc}/**": [
       "eslint"
+    ],
+    "**/lwc/**": [
+      "sfdx-lwc-jest --skipApiVersionCheck -- --bail --findRelatedTests"
```
Comment (pmdartus) [inline packages/templates/src/templates/project/package.json:44]: I am not a fan of running unit tests in a pre-commit hook. When working on a large feature is it fine to commit code that doesn't pass tests and fix the failing tests in a separate commit. On top of this, usually, a CI is in place to make sure all the tests are passing before merging a PR.

File: packages/templates/src/templates/project/package.json (hunk 2 above)
Comment (pmdartus) [reply packages/templates/src/templates/project/package.json:14]: > @pmdartus my understanding is that the LWC team will no longer release sfdx-lwc-jest new versions for each Salesforce release (npm indicates that the latest version is v0.10.4 for quite some time).

No it's the goal, we try to align `sfdx-lwc-jest` release with Salesforce release cadence.

File: packages/templates/src/templates/project/package.json (hunk 3 above)
Comment (pozil) [reply packages/templates/src/templates/project/package.json:44]: Ok, removing tests from pre-commit hook.

File: packages/templates/src/templates/project/package.json (hunk 2 above)
Comment (pozil) [reply packages/templates/src/templates/project/package.json:14]: Ok, removing the `--skipApiVersionCheck` flag now and waiting for a bug fix on the up-to-date version of `sfdx-lwc-jest`: https://github.com/salesforce/sfdx-lwc-jest/issues/221

File: packages/templates/src/generators/projectGenerator.ts [hunk 4]
```diff
@@ -111,6 +113,20 @@ export default class ProjectGenerator extends SfdxGenerator<ProjectOptions> {
 
     if (template === 'standard') {
       makeEmptyFolders(folderlayout, standardfolderarray);
+
```
Comment (smithgp) [inline packages/templates/src/generators/projectGenerator.ts:116]: @pozil Could these husky settings also get added to the `analytics` template section below? We're wanting to keep the npm & eslint related parts of that project type in sync with the `standard` template type.

File: packages/templates/src/templates/project/package.json (hunk 2 above)
Comment (pozil) [reply packages/templates/src/templates/project/package.json:14]: `sfdx-lwc-jest`'s bug is fixed, waiting for a new release.

File: packages/templates/src/generators/projectGenerator.ts (hunk 4 above)
Comment (pozil) [reply packages/templates/src/generators/projectGenerator.ts:116]: @smithgp good point, I've added the config to the the `analytics` template.

File: packages/templates/src/templates/project/package.json (hunk 2 above)
Comment (pozil) [reply packages/templates/src/templates/project/package.json]: Fixed: updated `sfdx-lwc-jest` to `0.13.0`.

**Comment by pozil** [comment]
All pending conversations are resolved, ready to resume review.

**Review by pmdartus — APPROVED** [review]

**Review by smithgp — APPROVED** [review]
Thanks for adding it to the analytics project!

**Review by brpowell — APPROVED** [review]

**Pull request merged**